src/modules/FaceDetection/FaceDetection.js

```diff
@@ -0,0 +1,278 @@
+    // calculate and return IoU
+    return overr * overc / (s1 * s1 + s2 * s2 - overr * overc);
+  }
+  /*
+		do clustering through non-maximum suppression
+	*/
+  const assignments = new Array(dets.length).fill(0);
+  const clusters = [];
+  for(let i = 0; i < dets.length; ++i)
+  {
+    // is this detection assigned to a cluster?
+    if(assignments[i] == 0)
+    {
+      // it is not:
+      // now we make a cluster out of it and see whether some other detections belong to it
+      let r = 0.0, c = 0.0, s = 0.0, q = 0.0, n = 0;
+      for(let j = i; j < dets.length; ++j)
+        if(calculate_iou(dets[i], dets[j]) > iouthreshold)
+        {
+          assignments[j] = 1;
+          r = r + dets[j][0];
+          c = c + dets[j][1];
+          s = s + dets[j][2];
+          q = q + dets[j][3];
+          n = n + 1;
+        }
+      // make a cluster representative
+      clusters.push([r / n, c / n, s / n, q]);
+    }
+  }
+
+  return clusters;
+};
+
+pico.instantiate_detection_memory = function(size)
+{
+  /*
+		initialize a circular buffer of `size` elements
+	*/
+  let n = 0;
+  const memory = [];
+  for(let i = 0; i < size; ++i)
+    memory.push([]);
+  /*
+		build a function that:
+		(1) inserts the current frame's detections into the buffer;
+		(2) merges all detections from the last `size` frames and returns them
+	*/
+  function update_memory(dets)
+  {
+    memory[n] = dets;
+    n = (n + 1) % memory.length;
+    dets = [];
+    for(i = 0; i < memory.length; ++i)
+      dets = dets.concat(memory[i]);
+    //
+    return dets;
+  }
+  /*
+		we're done
+	*/
+  return update_memory;
+};
+
+function rgba_to_grayscale(rgba, nrows, ncols) {
+  var gray = new Uint8Array(nrows * ncols);
+  for(var r = 0; r < nrows; ++r)
+    for(var c = 0; c < ncols; ++c)
+    // gray = 0.2*red + 0.7*green + 0.1*blue
+      gray[r * ncols + c] = (2 * rgba[r * 4 * ncols + 4 * c + 0] + 7 * rgba[r * 4 * ncols + 4 * c + 1] + 1 * rgba[r * 4 * ncols + 4 * c + 2]) / 10;
+  return gray;
+}
+
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+
+module.exports = exports = function(pixels, cb){
+  const pixelSetter = require('../../util/pixelSetter.js');
+  var facefinder_classify_region = function(r, c, s, pixels, ldim) {return -1.0;};
+  var drawSide = function(startX, startY, endX, endY){
+    for (var n = startX; n <= endX + 1; n++){
+      for (var k = startY; k <= endY + 1; k++){
+    
+        pixelSetter(n, k, [25, 25, 25, 255], pixels); // To remove 4th channel - pixels.set(n, k, 3, color[3]);
+      }
+    }
+  };
+  var cascadeurl = 'https://raw.githubusercontent.com/nenadmarkus/pico/c2e81f9d23cc11d1a612fd21e4f9de0921a5d0d9/rnt/cascades/facefinder';
+  color = 'rgba(20,20,20,1)';
+  color = color.substring(color.indexOf('(') + 1, color.length - 1); // Extract only the values from rgba(_,_,_,_)
+  color = color.split(',');
+  fetch(cascadeurl).then(function(response) {
```

Comment:
Maybe because of this the tests are failing, also does it work in node?
You could either install fetch for node from npm or use axios.
why is this used anyways?

Reply:
Ya this could be the reason, I will try to fetch it locally rather than from internet. It is used to fetch the trained model file which would be used to infer the face location